### PR TITLE
Removing -it flag from docker run command in pongo.sh

### PR DESCRIPTION
**Pull request by C13490888**
Is the -it flag needed in the docker run command?

### CODE REVIEW

**Comment by r1ckr** [comment]
Glad to see this PR! The `-it` makes having pongo in a CI/CD pipeline complicated.

File: pongo.sh [hunk 1]
```diff
@@ -211,7 +211,7 @@ function get_version {
     get_image
   fi
 
-  VERSION=$(docker run -it --rm -e KONG_LICENSE_DATA "$KONG_IMAGE" "${cmd[@]}")
+  VERSION=$(docker run --rm -e KONG_LICENSE_DATA "$KONG_IMAGE" "${cmd[@]}")
```
Comment (hutchic) [inline pongo.sh]: will `VERSION` still have the value after this without the `-it`?

Would this work maybe?
```
USE_TTY="-t"
test -t 1 && USE_TTY="-it"
docker run ${USE_TTY} --rm
```

**Comment by Tieske** [comment]
I haven't gotten around to test this yet, preferably I'd like a complete example for setting it up using Travis CI (or any other).

@r1ckr @C13490888 would you mind sharing your CI set up? or even propose a PR with the examples documented?

File: pongo.sh (hunk 1 above)
Comment (Tieske) [reply pongo.sh]: just tested this, and it works for me, dropping the whole `-it`, and it still properly populates the `$VERSION` variable

**Comment by r1ckr** [comment]
Would you like for me to update the documentation? The difference is that now we don't need to provide the `-it`. 

On our CD/CI we first we clone down the repo without the `-it` flag and then run it like this:
```
KONG_VERSION=${KONG_VERSION} pongo run ./spec
```
Pretty elegant 👍

**Comment by Tieske** [comment]
A PR with some piece of sample CI setup for "inspiration" would be nice, a couple of clone commands and a matrix to test it. A contribution would be great.

(and just last week we added `pongo lint` which should be in any CI as well 😄 )

**Comment by Tieske** [comment]
@r1ckr another question regarding the set up; do you cache the generated docker-images?

**Comment by r1ckr** [comment]
Hey @Tieske:

> @r1ckr another question regarding the set up; do you cache the generated docker-images?

We do not generate a docker image with the plugin we are creating, we test it with pongo and later mount the volume onto the upstream docker container, the pipeline looks like this:
```
##### 1. Set Kong version #####
KONG_VERSION=1.2.1

##### 2. Run our tests for the plugin #####
KONG_VERSION=${KONG_VERSION} pongo run ./spec

##### 3. Run kong mounting the plugin #####
docker run -d --name kong \
-e KONG_LUA_PACKAGE_PATH="/custom/?.lua;;"
-e KONG_PLUGINS="bundled,my-plugin-dir"
-v $(pwd)/my-plugin-dir:/custom/kong/plugins/my-plugin-dir \
kong:${KONG_VERSION}
```

> A PR with some piece of sample CI setup for "inspiration" would be nice, a couple of clone commands and a matrix to test it. A contribution would be great.
> 
> (and just last week we added `pongo lint` which should be in any CI as well 😄 )

I could work on the sample CI setup and the clone commands, what do you mean by the matrix to test it? Would you have an example?

**Comment by Tieske** [comment]
> I could work on the sample CI setup and the clone commands, what do you mean by the matrix to test it? Would you have an example?

test matrix, which versions of Kong, what combinations of datastores (if relevant)
eg. `1.3.0 + postgres`, `1.3.0 + cassandra`, `1.4.2 + postgres`, etc.

> We do not generate a docker image with the plugin we are creating, we test it with pongo and later mount the volume onto the upstream docker container

I meant whether you're caching the images that Pongo itself creates (since they are static for each Kong version, but never exist on a CI vm, hence always get re-created). But your response already provides me the answer 😄 .

From your example I get you're only testing against a single Kong version, it would probably make more sense when you'd test against multiple Kong versions

**Comment by Tieske** [comment]
@r1ckr here's what I meant with the matrix: https://docs.travis-ci.com/user/build-matrix/

**Comment by r1ckr** [comment]
> From your example I get you're only testing against a single Kong version, it would probably make more sense when you'd test against multiple Kong versions

This is a good advice and caching will be essential. Let me come back to you with that matrix.

**Comment by Tieske** [comment]
I read something about caching docker images somewhere, let me check where...

Here it is: https://github.com/travis-ci/travis-ci/issues/5358

Especially this comment seems useful, since when testing against multiple Kong images, most of the images will be shared: https://github.com/travis-ci/travis-ci/issues/5358#issuecomment-519389443

**Comment by Tieske** [comment]
fyi; today added CI setup to Pongo; see https://github.com/Kong/kong-pongo#setting-up-ci

**Pull request merged**